darts/models/ensemble_model.py

```diff
@@ -23,50 +24,114 @@ class EnsembleModel(ForecastingModel):
+                 "All models are of type darts.models.ForecastingModel which do not support covariates.",
+                 logger
+                 )
+        raise_if(not self.is_global_ensemble and past_covariates is not None,
+                 "All models are of type darts.models.ForecastingModel which do not support covariates.",
+                 logger
+                 )
+
+        self.is_univariate = isinstance(series, TimeSeries)
+        if past_covariates is not None:
+            self.is_univariate_covariate = isinstance(past_covariates, TimeSeries)
+
+        raise_if(past_covariates is not None and (self.is_univariate != self.is_univariate_covariate),
+                 "Both series and covariates have to be either univariate or multivariate.",
+                 logger
+                 )
+
+        super().fit(series, past_covariates, future_covariates)
+
+    def _stack_ts_seq(self, seq1, seq2):
+        # stacks two sequences of timeseries elementwise
+        return [ts1.stack(ts2) for ts1, ts2 in zip(seq1, seq2)]
+
+    def _split_multi_ts_sequence(self, n: int, ts_sequence: Sequence[TimeSeries]
+                                 ) -> Tuple[Sequence[TimeSeries], Sequence[TimeSeries]]:
+        left = [ts[:-n] for ts in ts_sequence]
+        right = [ts[-n:] for ts in ts_sequence]
+        return left, right
 
     def predict(self,
                 n: int,
-                num_samples: int = 1) -> TimeSeries:
-        super().predict(n, num_samples)
+                series: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+                past_covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+                future_covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+                num_samples: int = 1,
+                ) -> Union[TimeSeries, Sequence[TimeSeries]]:
+
+        super().predict(n=n, series=series,
+                        past_covariates=past_covariates, future_covariates=future_covariates, num_samples=num_samples)
+
+        if self.is_global_ensemble and not self.is_univariate:
+            predictions = self.models[0].predict(n=n, series=series,
+                        past_covariates=past_covariates, future_covariates=future_covariates, num_samples=num_samples)
+        else:
+            predictions = self.models[0].predict(n=n, num_samples=num_samples)
 
-        predictions = self.models[0].predict(n=n, num_samples=num_samples)
         if len(self.models) > 1:
             for model in self.models[1:]:
-                predictions = predictions.stack(model.predict(n=n, num_samples=num_samples))
+                if self.is_global_ensemble and not self.is_univariate:
+                    prediction = model.predict(n=n, series=series,
+                        past_covariates=past_covariates, future_covariates=future_covariates, num_samples=num_samples)
+                    predictions = self._stack_ts_seq(predictions, prediction)
```

Comment:
I don't think I understand: Note that I'm not stacking everything into one timeseries, but for each training timeseries I'm building a stack and train regression model with it. So let's assume that we have M models in ensemble and N timeserie samples, than this would return a N-sequence of M-stacked predictions. And that would be used for training the regression model. 

As far as you told me about the single timeserie case, this is the way to do it: use M-stacked predictions as covariates. In this case simply N=1.

Am I missing something?

Reply:
No you are right. It was done like that for N=1 already before. I'm just thinking that it's not 100% satisfying because it is abusing the "component" dimension of the TimeSeries IMO. However it is convenient (although works naturally out of the box with RegressionModels, so let's leave it like that (at least for now).

Btw: do you know if it still works if the original series are multivariate?

Reply:
> I'm just thinking that it's not 100% satisfying because it is abusing the "component" dimension of the TimeSeries IMO and if you remember, it wasn't obvious for me at the beginning how it works.
> Btw: do you know if it still works if the original series are multivariate?
I never checked, but I don't think it will work tbh, because as you mentioned, we're already 'occupying' `component` dimension for another purpose.

Reply:
We should then at least test then that the target is not multivariate and raise an exception if it is, stating that Ensemble models do not yet support multivariate series.